validate_md_links.py

```diff
@@ -24,6 +24,12 @@ def toc_md_files(toc, dirs):
 
 assert anchor_re.findall('<a id="comparison-to-c"/>') == ['comparison-to-c']
 
+third_party_trademarks = {'synopsys', 'coverity'}
+
+third_party_trademark_re = re.compile(
+    r'(?i:(' + '|'.join(third_party_trademarks)
+    + r'))([*®™]|&reg;|&trade;|&copy;|\\\*)?')
```

Comment:
urgh, I can't read this. Can you document what you mean with a unit test? A simple `assert third_party_trademark_re.match(...) == ...` on top level is sufficient.

Also, if you want to capture this in the future, then I suggest allowlist rather than blocklist, i.e. complain if the word before `®` or `™` does *not* match smth like Intel, Simics, DML.

Reply:
If you want to protect from using external trademarks without `*`, then  you need a much bigger list, because the world has more than two trademarked things.

And if we are to make this kind of check, then it's better to do it globally for all simics docs.